src/utils.js

```diff
@@ -86,3 +86,58 @@ export function toDataBody(value) {
 export function qsify(obj) {
   return toQuerystring(JSON.parse(JSON.stringify(obj)));
 }
+
+/**
+ * Checks if a version is within the provided range.
+ *
+ * @param  {String} version The version to check.
+ * @param  {String} min     The minimum version (inclusive).
+ * @param  {String} max     The minimum version (exclusive).
+ * @throws {Error} If the version is outside of the provided range.
+ */
+export function checkVersion(version, min, max) {
+  const extract = (str) => str.split(".").map(x => parseInt(x, 10));
+  const [verMajor, verMinor] = extract(version);
+  const [minMajor, minMinor] = extract(min);
+  const [maxMajor, maxMinor] = extract(max);
+  const msg = `Version ${version} doesn't match ${min} <= x < ${max}`;
+  const checks = [
+    verMajor < minMajor,
+    verMajor === minMajor && verMinor < minMinor,
+    verMajor > maxMajor,
+    verMajor === maxMajor && verMinor >= maxMinor,
+  ];
+  if (checks.some(x => x)) {
+    throw new Error(msg);
+  }
+}
+
+/**
+ * Generates a decorator function ensuring a version check is performed against
+ * the provided requirements before executing it.
+ *
+ * @param  {String} min The required min version (inclusive).
+ * @param  {String} max The required max version (inclusive).
+ * @return {Function}
+ */
+export function support(min, max) {
+  return function(target, key, descriptor) {
+    const fn = descriptor.value;
+    return {
+      configurable: true,
+      get() {
+        const wrappedMethod = (...args) => {
+          return this.fetchHTTPApiVersion()
```

Comment:
This makes me realize the `support` decorator is used on `index.js` (the client), meaning that batch operations won't benefit from the check.

Should we do this on the request object so that batch operations also issue a warning ? (see #65 )